### DISCUSSION TITLE
Sort not working with managed property

### DISCUSSION THREAD

**Discussion by MiteshreeBarolia** [microsoft-search/pnp-modern-search#3404] [Q&A]
### Check existing issues before logging new ones
If you are not using the latest version, triage your issue with the latest version before logging as we won't provide patches to older versions.

### Issues related to data mapping, web part setup etc, and SharePoint columns to crawled properties to managed properties should be asked in the discussion section.

**For any question regarding Web Parts usage, please use the ["Discussions"](https://github.com/microsoft-search/pnp-modern-search/discussions) tab instead of creating a new issue.**.

__Remove the above section before submitting, and removal of the below template will result in the issue being closed.__

**Version used**
Ex: 4.8

**Describe the bug**
When we add the managed property name for sorting like date type or integer type it is not sorting the data as required. Checked settings for managed property as well and the sort option is active. But when we select the option for sorting it did not sort the data. When we try with the Title column as well, we get error. we are using version 4.8. 

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Go to managed property create date type managed property RefinableDate03.
2. Go to sort setting add the managed property.
3. Save and try to do sorting on the search results we see.
4. Select Title for one sort option as well and try to sort it by selecting the option. 
5. See error

**Expected behavior**
When we sort with the date one it should show the latest reviewed one and will be able to sort it. For title one it should come in alphabetic order. 

**Screenshots**

**Desktop (please complete the following information):**
 - Chrome

**Additional context**

**Comment by kasperbolarsen** [comment]
Re: RefinableDate03
Can you verify that RefinableDate03 is in the "Selected Properties" in the Results web part? ( and contains data)

Re: Title
Out of the box Title is not sortable. Have you mapped Title to a RefinableString?
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/60eb25a7-f7d7-4d98-b987-4471bcc076d5)

**Comment by MiteshreeBarolia** [comment]
Yes, RefinableDate03 this is added to the selected properties and contains data. The Title we are mapping in RefinableString and then putting in Sort list, but its not working as expected

**Comment by kasperbolarsen** [comment]
FYI the error message you are getting is not visible in this thread.

Can you share a screen shot of:
"Manage Columns"  ( which layout are you using?)
"Sort settings"

**Comment by MiteshreeBarolia** [comment]
![Sort Error](https://github.com/microsoft-search/pnp-modern-search/assets/153707254/61157d6b-0855-4553-aa55-24831548e34b)
RefinableString13 - mapped with ows_LinkTitle
RefinableDate03 - mapped with lastrevieweddate

After adding them for sort the search result did not get sort as required in increasing/decreasing order. The order is random. 
![Sort Settings](https://github.com/microsoft-search/pnp-modern-search/assets/153707254/1b288310-9ddc-42b8-bc6a-2c43eae21781)

**Comment by kasperbolarsen** [comment]
That 500 error is very often due to one of the properties not being sortable. 
Can you try to remove them all, check if it works,  then add one of the properties,  check again and so forth

**Comment by MiteshreeBarolia** [comment]
I tried by removing and adding one by one, but it does not work for any of it. The error goes away but it does not sort the data in ascending/descending set, it shows the result in random sequence.

**Comment by MiteshreeBarolia** [comment]
When i try with the column names present like filename/recentviewcount/created still the data is not getting sort

**Comment by MiteshreeBarolia** [comment]
We created an int type managed property and attached it to no of views and used it in sorting, now the sorting for this one is happening as expected and for a date one as well. But not able to achieve the Title one for all the lists and libraries.

**Reply by kasperbolarsen** [reply]
The Managed property Title isn't sortable. You will have to map the crawled properties of Title to a RefinableString in order to achieve this.

**Reply by MiteshreeBarolia** [reply]
We did the Title one through a Refinable string by mapping the same properties as of title in it and then used the refinable string for sorting

**Reply by kasperbolarsen** [reply]
And you are seeing values in the RefinableString using the debug layout or SP Editor / SharePoint Search Query tool ?
